src/com/google/enterprise/adaptor/Config.java

```diff
@@ -1014,7 +1014,7 @@ File getConfigFile() {
    * @return map of all endings of prefix split into key value pairs
    */
   public synchronized Map<String, String> getValuesWithPrefix(String prefix) {
-    Map<String, String> values = new HashMap<String, String>();
+    Map<String, String> values = new TreeMap<String, String>();
```

Comment:
I was thinking to try to avoid similar future problems.  Is there any benefit to leaving HashMap?